tsweb/tsweb.go

```diff
@@ -626,57 +634,43 @@ type errorHandler struct {
 
 // ServeHTTP implements the http.Handler interface.
 func (h errorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	logf := h.opts.Logf
-	if l := logger.LogfKey.Value(r.Context()); l != nil {
-		logf = l
-	}
-
 	// Keep track of whether a response gets written.
 	lw, ok := w.(*loggingResponseWriter)
 	if !ok {
-		lw = &loggingResponseWriter{
-			ResponseWriter: w,
-			logf:           logf,
-		}
+		lw = newLogResponseWriter(h.opts.Logf, w, r)
 	}
 
-	// In case the handler panics, we want to recover and continue logging the
-	// error before raising the panic again for the server to handle.
-	var panicRes any
+	var err error
 	defer func() {
-		if panicRes != nil {
-			panic(panicRes)
+		// In case the handler panics, we want to recover and continue logging
```

Comment:
Would it be easier to reason about if this functionality was in its own middleware? Or will it make things more difficult?

Reply:
I think that would make things more difficult, but I can't be sure we're imagining the same thing.